### PR TITLE
widget for sequence comparison

### PR DESCRIPTION
**Pull request by levinas**

### CODE REVIEW

File: kbase-extension/static/kbase/js/widgets/function_output/kbaseSeqCompView.js [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+            return this;
+        },
+
+        loggedOutCallback: function(event, auth) {
+            this.token = null;
+            this.render();
+            return this;
+        },
+
+        uuid: function() {
```
Comment (briehl) [inline kbase-extension/static/kbase/js/widgets/function_output/kbaseSeqCompView.js]: There's now a separate module, `util/string` you can use as a dependency that provides uuids. See https://github.com/kbase/narrative/blob/develop/docs/developer/widget_helper_modules.md#utilstring

File: kbase-extension/static/kbase/js/widgets/function_output/kbaseSeqCompView.js [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+            	}, function(data) {
+            	    container.empty();
+            	    container.append('<p>[Error] ' + data.error.message + '</p>');
+            	});
+            };
+            ready();
+            return this;
+        },
+
+        getData: function() {
```
Comment (briehl) [inline kbase-extension/static/kbase/js/widgets/function_output/kbaseSeqCompView.js]: The need for the `getData` function died with the old card landing page layout. Unless it's used internally, you don't need this function.

File: kbase-extension/static/kbase/js/widgets/function_output/kbaseSeqCompView.js [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+        ws_id: null,
+        ws_name: null,
+        token: null,
+        width: 1150,
+        options: {
+            ws_id: null,
+            ws_name: null
+        },
+        loadingImage: window.kbconfig.loading_gif,
+        wsUrl: window.kbconfig.urls.workspace,
```
Comment (briehl) [inline kbase-extension/static/kbase/js/widgets/function_output/kbaseSeqCompView.js]: Depending on the global window.kbconfig variable has been deprecated. There's now a helper module called `narrativeConfig` that you can use as a dependency to get urls and such (just add it to the list in the `define` statement). See here for a couple examples: https://github.com/kbase/narrative/blob/develop/docs/developer/widget_helper_modules.md#narrative-config

**Comment by levinas** [comment]
Thanks @briehl for your comments with examples. I've edited the code to use the narrative utils.

File: kbase-extension/static/kbase/js/widgets/function_output/kbaseSeqCompView.js [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+            this._super(options);
+
+            this.ws_name = options.ws_name;
+            this.ws_id = options.ws_id;
+            if (options.report_name && options.workspace_name) {
+                  this.ws_id = options.report_name;
+                  this.ws_name = options.workspace_name;
+            }
+	    this.ws_id = 'ecoli.dnadiff.output';
+	    this.ws_name = 'fangfang:1454986986211';
```
Comment (briehl) [inline kbase-extension/static/kbase/js/widgets/function_output/kbaseSeqCompView.js]: It looks like you're setting this.ws_id and this.ws_name here explicitly, which will override the user options...

**Comment by levinas** [comment]
I forgot to take out the hardcoded example. Now removed. Thanks for spotting that.

**Comment by briehl** [comment]
Yup, that's why it's nice to have extra eyes on things. :) merging.

**Pull request merged**